controller/controller.go

```diff
@@ -165,10 +164,18 @@ func (c *Impl) EnqueueLabelOf(namespaceLabel, nameLabel string) func(obj interfa
 				return
 			}
 
-			controllerKey = fmt.Sprintf("%s/%s", controllerNamespace, controllerKey)
-		}
+			c.EnqueueKey(fmt.Sprintf("%s/%s", controllerNamespace, controllerKey))
+			return
 
-		c.EnqueueKey(controllerKey)
+		}
+		// Pass through namespace of the object itself if no namespace label specified.
+		// This is for the scenario that object and the parent resource are of same namespace,
+		// e.g. to enqueue the reivison of an endpoint.
+		// Note that for objects whose parent resource are cluster-scoped, here it adds the
+		// namespace part that we don't indeed need. Callers should be able to call
```

Comment:
Perhaps we should treat this as the "namespaced" variety, and have a separate version for "cluster"-scoped?

I would expect that this would almost certainly have to be used with a GroupVersionKind-filtered function, so the logic setting it up should know which they are dealing with.